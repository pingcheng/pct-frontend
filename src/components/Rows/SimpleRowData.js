import React, { Component } from "react";
import PropTypes from "prop-types";

export class SimpleRowData extends Component {

	static propTypes = {
		label: PropTypes.string.isRequired
	};

	render() {
		return (
			<div className="flex mb-4">
				<div
					className="text-gray-400 w-1/3 text-right pr-2"
					style={{
						minWidth: "33.33333%"
					}}
				>{ this.props.label.trim().toUpperCase() }</div>

				<div
					className="flex-1 pl-2"
				>{ this.props.children }</div>
			</div>
		)
	}
}