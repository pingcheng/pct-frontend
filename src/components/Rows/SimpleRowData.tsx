import React, { ReactNode } from "react";

export type SimpleRowDataProps = {
  label: string;
  children?: ReactNode;
};

export function SimpleRowData({
  label,
  children,
}: SimpleRowDataProps): JSX.Element {
  return (
    <div className="flex mb-4">
      <div
        className="text-gray-400 w-1/3 text-right pr-2"
        style={{
          minWidth: "33.33333%",
        }}
      >
        {label.trim().toUpperCase()}
      </div>

      <div className="flex-1 pl-2">{children}</div>
    </div>
  );
}
