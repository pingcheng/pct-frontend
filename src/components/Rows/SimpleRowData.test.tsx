import { render, screen } from "@testing-library/react";
import { SimpleRowData, SimpleRowDataProps } from "./SimpleRowData";

describe("test <SimpleRowData />", () => {
  const props = {
    label: "Label",
  } as const;

  const renderComponent = (props: SimpleRowDataProps) =>
    render(<SimpleRowData {...props} />);

  it("should render correctly", () => {
    const { container } = renderComponent(props);
    expect(container.innerHTML).toMatchSnapshot();
  });

  it("should render label name", () => {
    renderComponent(props);
    expect(screen.getByText("LABEL")).toBeInTheDocument();
  });

  it("should render children content", () => {
    renderComponent({
      ...props,
      children: "Children Text",
    });
    expect(screen.getByText("Children Text")).toBeInTheDocument();
  });
});
