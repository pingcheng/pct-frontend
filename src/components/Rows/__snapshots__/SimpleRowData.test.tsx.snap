// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <SimpleRowData /> should render correctly 1`] = `"<div class=\\"flex mb-4\\"><div class=\\"text-gray-400 w-1/3 text-right pr-2\\" style=\\"min-width: 33.33333%;\\">LABEL</div><div class=\\"flex-1 pl-2\\"></div></div>"`;
